cirq/linalg/transformations.py

```diff
@@ -326,6 +324,10 @@ def partial_trace(tensor: np.ndarray, keep_indices: List[int]) -> np.ndarray:
     return np.einsum(tensor, left_indices + right_indices)
 
 
+class UnfactorizableStateVectorError(ValueError):
```

Comment:
AFAICT, this is raised when the input state $|\psi_{AB}\rangle$ viewed as a state of two subsystems A and B (each possibly consisting of multiple qubits) is entangled (i.e. not a product state or IOW if $|\psi_{AB}\rangle \ne |\phi_A\rangle\otimes|\gamma_B\rangle$).

A better name for this exception is "EntangledStateError". Also, docstring would be nice.

Reply:
done